src/main-page.ts

```diff
@@ -60,6 +61,9 @@ export class MainPage implements Page {
     }
 
     private onClickModalReady(ev: ButtonMouseEvent) {
+        const wakeLock = new WakeLockManager();
+        wakeLock.requestWakeLock();
```

Comment:
Why do you create a new wakeLockManager instance every time user clicks on this button? Shouldn't you create a single instance of it and re-use it across the app.